provisioning/transport/http/src/ProvisioningTransportHandlerHttp.cs

```diff
@@ -148,10 +148,43 @@ public ProvisioningTransportHandlerHttp()
 
                     cancellationToken.ThrowIfCancellationRequested();
 
-                    operation = await client.RuntimeRegistration.OperationStatusLookupAsync(
+                    try
+                    {
+                        operation = await client.RuntimeRegistration.OperationStatusLookupAsync(
                         registrationId,
                         operationId,
                         message.IdScope).ConfigureAwait(false);
+                    }
+                    catch(HttpOperationException oe)
+                    {
+                        bool isTransient = oe.Response.StatusCode >= HttpStatusCode.InternalServerError
+                            || (int)oe.Response.StatusCode == 429;
+                        try
+                        {
+                            var errorDetails = JsonConvert.DeserializeObject<ProvisioningErrorDetailsHttp>(oe.Response.Content);
+                            if (isTransient)
+                            {
+                                serviceRecommendedDelay = errorDetails.RetryAfter;
```

Comment:
errorDetails will not have RetryAfter header for status code 5xx, those cases should set to the default retry after delay instead.

Reply:
It will be defaulted to s_defaultOperationPoolingIntervalMilliseconds at line 147